src/ray/raylet/scheduling_resources.h

```diff
@@ -264,8 +324,9 @@ class ResourceIdSet {
   /// \brief Return a set of resource IDs.
   ///
   /// \param resource_id_set The resource IDs to return.
+  /// \param strict If set to to true, creates any resources that do not already exist in the ResourceIdSet. Else ignores any new resources.
```

Comment:
Why would we ever want to ignore new resources?

Reply:
The resource might have been deleted while it was acquired by some task, and thus a release should not add them back to `available_resources_`.